### DISCUSSION TITLE
DataGrid RowSelect

### DISCUSSION THREAD

**Discussion by franklupo** [microsoft/fluentui-blazor#2422] [General]
`DataGrid.OnRowClick` sends event even when SelectColumn is clicked. This is not normal behavior. All SelectColumn events should be handled by it only. How do I solve this?

**Comment by dvoituron** [comment]
In the `SelectColumn`, using the `SelectFromEntireRow="false"` attribute, the `OnRowClick` event is still triggered.
But you will have the `OnCellClick` event to detect which cell is clicked `e.GridColumn`.
I think you could use this event to detect which column to exclude.

```xml
<FluentDataGrid Items="@People" ShowHover="@SelectFromEntireRow" TGridItem="Person"
                OnRowClick="@(e => Console.WriteLine($"Row {e.RowIndex} clicked"))"
                OnCellClick="@(e => Console.WriteLine($"Cell on column {e.GridColumn} clicked"))">
```

```
Cell on column 2 clicked
Row 4 clicked

Cell on column 1 clicked
Row 4 clicked
```

**Reply by franklupo** [reply]
I did this but it's horrible
